pkg/shell/index.js

```diff
@@ -217,20 +220,34 @@ define([
 
     /* When the machine list is ready we start processing navigation */
     $(machines)
+       .on("read-complete", function(ev) {
```

Comment:
Ready gets fired the moment the 1st message comes in. Read complete is when processing triggered by a file change completes.